File: problems/6. Zigzag Conversion/main.py
class Solution(object):
    def convert(self, s, numRows):
        """
        :type s: str
        :type numRows: int
        :rtype: str
        """
        if numRows == 1 or numRows >= len(s):
            return s

        results = [''] * numRows
        row = 0
        change_direction = False

        for c in s:
            results[row] += c
            if row == 0 or row == numRows - 1:
                change_direction = not change_direction
            row += 1 if change_direction else -1

        return ''.join(results)

File: problems/6. Zigzag Conversion/main.js
/**
 * @param {string} s
 * @param {number} numRows
 * @return {string}
 */
var convert = function (s, numRows) {
  if (numRows === 1 || numRows >= s.length) {
    return s;
  }

  const results = Array(numRows).fill("");
  let row = 0;
  let changeDirection = false;

  for (const c of s) {
    results[row] += c;
    if (row === 0 || row === numRows - 1) {
      changeDirection = !changeDirection;
    }
    row += changeDirection ? 1 : -1;
  }

  return results.join("");
};
